sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -1191,6 +1235,22 @@ class HistGradientBoostingRegressor(RegressorMixin, BaseHistGradientBoosting):
 
         .. versionadded:: 0.23
 
+    interaction_cst : iterable of iterables of int, default=None
+        Specify interaction constraints, i.e. sets of features which can
+        only interact with each other in child nodes splits.
+
+        Each iterable materializes a constraint by the set of indices of
+        the features that are allowed to interact with each other.
+        If there are more features than specified in these constraints,
+        they are treated as if they were specified as an additional set.
+
+        For instance, with 5 features in total, `interaction_cst=[{0, 1}]`
+        is equivalent to `interaction_cst=[{0, 1}, {2, 3, 4}]`,
+        and specifies that a single branch of a tree will either only split
+        on features 0 and 1 or only split on features 2, 3 and 4.
```

Comment:
Is my understanding exact?
```suggestion
        and specifies that each single branch of a tree will either be able
        to only split on features 0 and 1 or only split on features 2, 3 and 4.
```

Reply:
I went with
```
specifies that each branch of a tree will either only split
        on features 0 and 1 or only split on features 2, 3 and 4.
```
The "being able to split" seems overcomplicated to me. It's not the ability that counts, but the fact that it does.